sklearn/metrics/tests/test_classification.py

```diff
@@ -1381,117 +1393,167 @@ def test_precision_recall_f1_no_labels_average_none():
     # |y_i| = [0, 0, 0]
     # |y_hat_i| = [0, 0, 0]
 
-    p, r, f, s = assert_warns(UndefinedMetricWarning,
-                              precision_recall_fscore_support,
-                              y_true, y_pred, average=None, beta=beta)
-    assert_array_almost_equal(p, [0, 0, 0], 2)
-    assert_array_almost_equal(r, [0, 0, 0], 2)
-    assert_array_almost_equal(f, [0, 0, 0], 2)
+    func = precision_recall_fscore_support
+    my_assert = (assert_warns if zero_division == "warn"
```

Comment:
if you must do something like this, use functools.partial to capture the arguments too.

But I think tests must be very readable code, as the reader needs to be absolutely certain of their correctness to be confident that they in turn imply the corretness of the code.

Reply:
simplified in the new version with two separated tests